### PR TITLE
Support .yaml extension for meta file

### PR DESCRIPTION
**Pull request by jtyr**
##### SUMMARY
Galaxy Portal supports import of roles which contain `main.yaml` meta file since [this](https://github.com/ansible/galaxy/pull/19) PR. Unfrotunately `ansible-galaxy` command doesn't support it yet and end up with the following error:

```
$ ansible-galaxy install jtyr.ansible_docker_swarm      
- downloading role 'ansible_docker_swarm', owned by jtyr
- downloading role from https://github.com/jtyr/ansible-docker_swarm/archive/master.tar.gz
 [WARNING]: - jtyr.ansible_docker_swarm was NOT installed successfully: this role does not appear to have a
meta/main.yml file.

ERROR! - you can use --ignore-errors to skip failed roles and finish processing the list.
```

This PR is adding this support which makes installation of roles containing the `main.yaml` meta file possible:

```
$ ansible-galaxy install jtyr.ansible_docker_swarm
- downloading role 'ansible_docker_swarm', owned by jtyr
- downloading role from https://github.com/jtyr/ansible-docker_swarm/archive/master.tar.gz
- extracting jtyr.ansible_docker_swarm to /home/jtyr/.ansible/roles/jtyr.ansible_docker_swarm
- jtyr.ansible_docker_swarm (master) was installed successfully
```

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
`ansible-galaxy`

##### ANSIBLE VERSION
```
ansible 2.6.4
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.7.0 (default, Sep 15 2018, 19:13:07) [GCC 8.2.1 20180831]
```

##### ADDITIONAL INFORMATION
None

### CODE REVIEW

**Comment by samdoran** [comment]
cc @alikins @chouseknecht

**Review by maxamillion — APPROVED** [review]

File: lib/ansible/galaxy/role.py [hunk 1]
```diff
@@ -96,16 +96,17 @@ def metadata(self):
...
-                try:
-                    f = open(meta_path, 'r')
-                    self._metadata = yaml.safe_load(f)
-                except:
-                    display.vvvvv("Unable to load metadata for %s" % self.name)
-                    return False
-                finally:
-                    f.close()
+            for meta_main in self.META_MAIN:
+                meta_path = os.path.join(self.path, meta_main)
```
Comment (bcoca) [inline lib/ansible/galaxy/role.py:100]: i would just add 'meta' here  and do 1 join vs 2 of em

**Review by bcoca — APPROVED** [review]
comment is not a blocker, lgtm as is, but would prefer 1 less join operation

**Comment by chouseknecht** [comment]
LGTM. This change matches what we're doing on the server side: https://github.com/ansible/galaxy/blob/devel/galaxy/importer/loaders/role.py#L35

**Comment by alikins** [comment]
If there are existing galaxy roles that have main.yaml, this seems OK to me.

For cases where there is a main.yml and a main.yaml, should there be an error? (ditto for galaxy side import as well).

File: lib/ansible/galaxy/role.py (hunk 1 above)
Comment (jtyr) [reply lib/ansible/galaxy/role.py:100]: That would require to add another `join` on the line 273. I think it has better readability as it's now.

**Comment by jtyr** [comment]
> For cases where there is a main.yml and a main.yaml, should there be an error?

There should not be an error if the order of reading the files is consistent in Galaxy as well as in Ansible. Currently it's consistent in both by trying to read `.yml` and then `.yaml`.

**Comment by maxamillion** [comment]
rebuild_merge

**Pull request merged**